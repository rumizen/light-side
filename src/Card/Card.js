import React, { Component } from 'react';
import '../images/star.png';
import '../images/starfav.png';
import './Card.scss';
import Snippet from '../Snippet/Snippet.js';


class Card extends Component {
  constructor() {
    super();
    // this.state = {
    //   isFavorited: false
    // };
  }

  favoriteCard = () => {
    this.setState({ isFavorited: !this.state.isFavorited })
    setTimeout(() => {
      this.props.updateFavorites(
        { ...this.props, isFav: this.state.isFavorited },
        this.state.isFavorited
      );
    })
  }

  render() {
    const snippets = Object.keys(this.props).map(snippet => {
      return <Snippet
        category={snippet}
        text={this.props[snippet]}
        />
    });

    return (
      <section className="card" id={Date.now()} key={Date.now()}>
        <h2>{this.props.name}</h2>
        {snippets}
        <div onClick={e => this.props.updateFavorite(this.props.Name)}>
          {!this.props.isFav && (
            <img
              src={require("../images/star.png")}
              alt=""
              className="fav-icon"
            />
          )}
          {this.props.isFav && (
            <img
              src={require("../images/starfav.png")}
              alt=""
              className="fav-icon"
            />
          )}
        </div>
      </section>
    );
  };
};

export default Card;