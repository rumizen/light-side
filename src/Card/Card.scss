.card {
  background-color: rgb(77, 127, 202);
  border: solid 2px;
  box-shadow: 6px 5px 5px white;
  margin: 8px;
  width: 18%;
}

.fav-icon {
  height: 30px;
}