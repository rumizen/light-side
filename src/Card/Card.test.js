import React from "react";
import { shallow, mount } from "enzyme";
import Card from "./Card";

describe("Card", () => {
  it("should match the card snapshot", () => {
    const wrapper = shallow(
      <Card
        name="luke"
        gender="male"
        eyeColor="blue"
        height={6}
        birthYear={1992}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
