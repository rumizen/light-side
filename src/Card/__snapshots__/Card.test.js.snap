// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the card snapshot 1`] = `ReactWrapper {}`;

exports[`Card should match the snapshot when favorited 1`] = `ReactWrapper {}`;
